File: BodyMICalc/app/src/main/java/com/s16665/bodymicalc/ChartFragment.java
package com.s16665.bodymicalc;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.graphics.Color;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import java.util.ArrayList;
import java.util.List;

import com.github.mikephil.charting.charts.PieChart;
import com.github.mikephil.charting.components.Legend;
import com.github.mikephil.charting.data.PieData;
import com.github.mikephil.charting.data.PieDataSet;
import com.github.mikephil.charting.data.PieEntry;
import com.github.mikephil.charting.formatter.PercentFormatter;
import com.github.mikephil.charting.utils.ColorTemplate;


public class ChartFragment extends Fragment {
    PieChart pieChart;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.chart_fragment, container, false);
        pieChart = v.findViewById(R.id.piechart);
        setupPieChart();
        loadPieChartData();

        return v;
    }

    private void setupPieChart() {
        pieChart.setDrawHoleEnabled(true);
        pieChart.setUsePercentValues(false);
        pieChart.setEntryLabelTextSize(16);
        pieChart.setEntryLabelColor(Color.BLACK);
        pieChart.setCenterText("Ludzie z różnych krajów vs COVID");
        pieChart.setCenterTextSize(20);
        pieChart.getDescription().setEnabled(false);

        Legend legend = pieChart.getLegend();
        legend.setOrientation(Legend.LegendOrientation.VERTICAL);
        legend.setHorizontalAlignment(Legend.LegendHorizontalAlignment.LEFT);
        legend.setVerticalAlignment(Legend.LegendVerticalAlignment.TOP);
        legend.setDrawInside(false);
        legend.setEnabled(true);
    }

    private void loadPieChartData() {
        List<PieEntry> entries = new ArrayList<>();
        entries.add(new PieEntry(2000, "Zaszczepieni"));
        entries.add(new PieEntry(5000, "Zarażonych"));
        entries.add(new PieEntry(3000, "Zmarło"));
        entries.add(new PieEntry(1000, "Ozdrowieni"));

        ArrayList<Integer> colors = new ArrayList<>();
        for (int color : ColorTemplate.MATERIAL_COLORS) {
            colors.add(color);
        }

        for (int color : ColorTemplate.VORDIPLOM_COLORS) {
            colors.add(color);
        }

        PieDataSet dataSet = new PieDataSet(entries, "Ludzie z różnych krajów vs COVID");
        dataSet.setColors(colors);

        PieData data = new PieData(dataSet);
        data.setDrawValues(true);
        data.setValueFormatter(new PercentFormatter(pieChart));
        data.setValueTextSize(16);
        data.setValueTextColor(Color.WHITE);

        pieChart.setData(data);
        pieChart.invalidate();

    }
}

File: BodyMICalc/app/src/main/java/com/s16665/bodymicalc/ChartFragment.kt
package com.s16665.bodymicalc

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.github.mikephil.charting.charts.PieChart
import com.github.mikephil.charting.components.Legend
import com.github.mikephil.charting.data.PieData
import com.github.mikephil.charting.data.PieDataSet
import com.github.mikephil.charting.data.PieEntry
import com.github.mikephil.charting.formatter.PercentFormatter
import com.github.mikephil.charting.utils.ColorTemplate
import java.util.*

class ChartFragment : Fragment() {
    var pieChart: PieChart? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.chart_fragment, container, false)
        pieChart = v.findViewById(R.id.piechart)
        setupPieChart()
        loadPieChartData()
        return v
    }

    private fun setupPieChart() {
        pieChart?.setDrawHoleEnabled(true)
        pieChart?.setUsePercentValues(false)
        pieChart?.setEntryLabelTextSize(16f)
        pieChart?.setEntryLabelColor(Color.BLACK)
        pieChart?.setCenterText("Ludzie z różnych krajów vs COVID")
        pieChart?.setCenterTextSize(20f)
        pieChart?.getDescription()?.isEnabled = false
        val legend = pieChart?.getLegend()
        legend?.orientation = Legend.LegendOrientation.VERTICAL
        legend?.horizontalAlignment = Legend.LegendHorizontalAlignment.LEFT
        legend?.verticalAlignment = Legend.LegendVerticalAlignment.TOP
        legend?.setDrawInside(false)
        legend?.isEnabled = true
    }

    private fun loadPieChartData() {
        val entries: MutableList<PieEntry?> = ArrayList()
        entries.add(PieEntry(2000f, "Zaszczepieni"))
        entries.add(PieEntry(5000f, "Zarażonych"))
        entries.add(PieEntry(3000f, "Zmarło"))
        entries.add(PieEntry(1000f, "Ozdrowieni"))
        val colors = ArrayList<Int?>()
        for (color in ColorTemplate.MATERIAL_COLORS) {
            colors.add(color)
        }
        for (color in ColorTemplate.VORDIPLOM_COLORS) {
            colors.add(color)
        }
        val dataSet = PieDataSet(entries, "Ludzie z różnych krajów vs COVID")
        dataSet.colors = colors
        val data = PieData(dataSet)
        data.setDrawValues(true)
        data.setValueFormatter(PercentFormatter(pieChart))
        data.setValueTextSize(16f)
        data.setValueTextColor(Color.WHITE)
        pieChart?.setData(data)
        pieChart?.invalidate()
    }
}